packages/pg/lib/client.js

```diff
@@ -156,32 +153,56 @@ class Client extends EventEmitter {
     }
     let minConnectionCount = Number.MAX_VALUE
     let leastLoadedHosts = []
-    let hosts = hostsList.keys()
-    for (let value of hosts) {
-      let host = value
-      if (this.connectionParameters.topologyKeys !== '') {
+    for (var i = 1; i <= Client.topologyKeyMap.size; i++) {
+      let hosts = hostsList.keys()
+      for (let value of hosts) {
```

Comment:
You can take the the Hosts for this placementInfo from`placementInfoHostMap ` and then iterate over them?

Reply:
As agreed over call, since there can be multiple placementInfo for a particular preferenceValue both the logics will be similarly efficient so keeping this same as it builds on the current existing code.